From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage due to snd_usb_audio_quirk profile_name addition
Patch-mainline: Never, kABI fix
References: bsc#1091678

The addition of profile_name field to snd_usb_audio_quirk broke the kABI
as the USB-MIDI function is exposed for other drivers.  It's used only
in the standard usb-audio driver, so it's safe to move and hide it
with __GENKSYMS__ as usual.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/usb/usbaudio.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/sound/usb/usbaudio.h
+++ b/sound/usb/usbaudio.h
@@ -112,10 +112,12 @@ enum quirk_type {
 struct snd_usb_audio_quirk {
 	const char *vendor_name;
 	const char *product_name;
-	const char *profile_name;	/* override the card->longname */
 	int16_t ifnum;
 	uint16_t type;
 	const void *data;
+#ifndef __GENKSYMS__
+	const char *profile_name; 	/* override the card->longname */
+#endif
 };
 
 #define combine_word(s)    ((*(s)) | ((unsigned int)(s)[1] << 8))
